### ISSUE TITLE
Support mutli-repo in focus mode sidebar webview

### ISSUE THREAD

**Issue by brianjmurrell** [microsoft/vscode-pull-request-github#4988]
Type: <b>Bug</b>

In all but one of my repos, when I go to the GH PR extension and open a PR waiting for my review, I can then open the description page for it and click Check Out.  Once I have done that, I am given a pane in the left-hand pane which includes a comment box/button and a list of reviewers.

In one particular repo, I don't get this comment box/button/list of reviewers pane when I Check Out a PR waiting on my review.

The OUTPUT tab for _GitHub Pull Request_ displays the following when I click Check Out:

```
2023-06-30 14:11:21.706 [info] Review> Switch to Pull Request #12561 - start
2023-06-30 14:11:25.589 [info] Review> Switch to Pull Request #12561 - done
2023-06-30 14:11:25.590 [info] Review> Validate state in progress
2023-06-30 14:11:25.590 [info] Review> Validating state...
2023-06-30 14:11:25.642 [info] Found GitHub remote for folder /home/brian/daos/daos
2023-06-30 14:11:25.660 [info] Trying to use globalState for mentionable users.
2023-06-30 14:11:25.868 [info] No globalState for mentionable users.
2023-06-30 14:11:25.983 [info] Review> current branch jemalmbe/sre-1904 is associated with pull request #12561
2023-06-30 14:11:25.983 [info] Review> Resolving pull request
2023-06-30 14:11:26.930 [info] Review> Fetching pull request data
2023-06-30 14:11:26.930 [info] PullRequestTree> Adding PR #12561 to tree
2023-06-30 14:11:26.954 [info] Review> Register comments provider
2023-06-30 14:11:27.036 [info] PullRequestTree> Creating file and commit nodes for PR #384
2023-06-30 14:11:27.037 [info] PullRequestTree> Creating file and commit nodes for PR #12561
2023-06-30 14:11:27.040 [info] PullRequestTree> Getting children for Files node
2023-06-30 14:11:27.040 [info] PullRequestTree> Got all children for Files node
2023-06-30 14:11:27.041 [info] PullRequestTree> Getting children for Files node
2023-06-30 14:11:27.041 [info] PullRequestTree> Got all children for Files node
2023-06-30 14:11:27.723 [info] PullRequestTree> Local files have changed, refreshing Files node
2023-06-30 14:11:27.865 [info] PullRequestTree> Getting children for Files node
2023-06-30 14:11:27.865 [info] PullRequestTree> Got all children for Files node
2023-06-30 14:11:27.871 [info] PullRequestTree> Review threads have changed, refreshing Files node
2023-06-30 14:11:27.872 [info] PullRequestTree> Review threads have changed, refreshing Commits node
2023-06-30 14:11:27.872 [info] Review> Display pull request status bar indicator and refresh pull request tree view.
2023-06-30 14:11:27.873 [info] Review> Using focus mode = true.
2023-06-30 14:11:27.873 [info] Review> State validation silent = true.
2023-06-30 14:11:27.873 [info] Review> PR show should show = false.
2023-06-30 14:11:28.073 [info] PullRequestTree> Getting children for Files node
2023-06-30 14:11:28.073 [info] PullRequestTree> Got all children for Files node
2023-06-30 14:11:28.138 [info] PullRequestTree> Comments have changed, refreshing Files node
2023-06-30 14:11:28.139 [info] PullRequestTree> Comments have changed, refreshing Commits node
2023-06-30 14:11:28.139 [info] PullRequestTree> Review threads have changed, refreshing Files node
2023-06-30 14:11:28.139 [info] PullRequestTree> Review threads have changed, refreshing Commits node
2023-06-30 14:11:28.296 [info] PullRequestTree> Getting children for Files node
2023-06-30 14:11:28.297 [info] PullRequestTree> Got all children for Files node
2023-06-30 14:11:28.373 [info] PullRequestTree> Getting children for Files node
2023-06-30 14:11:28.373 [info] PullRequestTree> Got all children for Files node
2023-06-30 14:11:28.613 [info] PullRequestTree> Review threads have changed, refreshing Files node
2023-06-30 14:11:28.613 [info] PullRequestTree> Review threads have changed, refreshing Commits node
2023-06-30 14:11:28.616 [info] PullRequestTree> Getting children for Files node
2023-06-30 14:11:28.616 [info] PullRequestTree> Got all children for Files node
2023-06-30 14:12:24.396 [info] PullRequestTree> Local files have changed, refreshing Files node
2023-06-30 14:12:24.445 [info] PullRequestTree> Getting children for Files node
2023-06-30 14:12:24.446 [info] PullRequestTree> Got all children for Files node
2023-06-30 14:12:24.697 [info] PullRequestTree> Comments have changed, refreshing Files node
2023-06-30 14:12:24.697 [info] PullRequestTree> Comments have changed, refreshing Commits node
2023-06-30 14:12:24.697 [info] PullRequestTree> Review threads have changed, refreshing Files node
2023-06-30 14:12:24.697 [info] PullRequestTree> Review threads have changed, refreshing Commits node
2023-06-30 14:12:24.701 [info] PullRequestTree> Getting children for Files node
2023-06-30 14:12:24.701 [info] PullRequestTree> Got all children for Files node
2023-06-30 14:12:24.894 [info] PullRequestTree> Local files have changed, refreshing Files node
2023-06-30 14:12:25.097 [info] PullRequestTree> Getting children for Files node
2023-06-30 14:12:25.097 [info] PullRequestTree> Got all children for Files node
2023-06-30 14:12:25.097 [info] PullRequestTree> Getting children for Files node
2023-06-30 14:12:25.097 [info] PullRequestTree> Got all children for Files node
2023-06-30 14:12:25.251 [info] PullRequestTree> Comments have changed, refreshing Files node
2023-06-30 14:12:25.251 [info] PullRequestTree> Comments have changed, refreshing Commits node
2023-06-30 14:12:25.254 [info] PullRequestTree> Review threads have changed, refreshing Files node
2023-06-30 14:12:25.254 [info] PullRequestTree> Review threads have changed, refreshing Commits node
2023-06-30 14:12:25.420 [info] PullRequestTree> Getting children for Files node
2023-06-30 14:12:25.420 [info] PullRequestTree> Got all children for Files node
2023-06-30 14:12:25.638 [info] PullRequestTree> Getting children for Files node
2023-06-30 14:12:25.638 [info] PullRequestTree> Got all children for Files node
2023-06-30 14:12:25.881 [info] PullRequestTree> Review threads have changed, refreshing Files node
2023-06-30 14:12:25.881 [info] PullRequestTree> Review threads have changed, refreshing Commits node
2023-06-30 14:12:25.883 [info] PullRequestTree> Getting children for Files node
2023-06-30 14:12:25.883 [info] PullRequestTree> Got all children for Files node
```

Extension version: 0.66.2
VS Code version: Code 1.79.2 (695af097c7bd098fbf017ce3ac85e09bbc5dda06, 2023-06-14T08:59:55.818Z)
OS version: Linux x64 6.3.4-201.fc38.x86_64
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-10850H CPU @ 2.70GHz (12 x 2404)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|3, 13, 10|
|Memory (System)|62.44GB (25.05GB free)|
|Process Argv|--unity-launch --crash-reporter-id d33a766b-28e9-45e5-942d-3a9085ddc0c0|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|gnome|
|XDG_CURRENT_DESKTOP|GNOME|
|XDG_SESSION_DESKTOP|gnome|
|XDG_SESSION_TYPE|x11|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstes627:30244334
vslsvsres303:30308271
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscod805:30301674
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
pythonvs932:30410667
py29gd2263:30776702
vsclangdf:30486550
c4g48928:30535728
dsvsc012:30540252
pynewext54:30695312
azure-dev_surveyone:30548225
vsccc:30610678
282f8724:30602487
pyind779:30671433
f6dab269:30613381
pythonsymbol12:30671437
2i9eh265:30646982
showlangstatbar:30737416
vsctsb:30748421
pythonms35:30701012
03d35959:30757346
pythonfmttext:30731395
pythoncmv:30756943
fixshowwlkth:30771522
hideindicator:30766889
pythongtdpath:30769146
i26e3531:30780429
pythonnosmt12:30779714
pythonidxptcf:30772540
pythonnoceb:30776495

```

</details>

**Comment by alexr00** [comment]
@brianjmurrell thanks for including logs! 

It looks like the workspace you have open actually has two GitHub repositories and both of those repositories have PRs open from the branch that you have checked out, respectively:
- For repo A, you have PR `#384` checked out
- For repo B, you have PR `#12561` checked out

The sidebar view that you are describing with the comment box and the reviewers does not have much space, so we only show it when you have only 1 PR checked out. You can get the same functionality that's in that view from the PR description.

**Comment by alexr00** [comment]
Using the same branch name shouldn't make any difference at all. It's the fact that you are using a multi-repo workspace and you have a PR checked out in each repo.

You can get the same functionality that's available in the sidebar from the PR description. 

> Could the sidebar view have expanders for the checked out PRs in the same was as the source view, and many other views have?

Yes we could do something like this. I can re-open this issue and re-purpose it as a feature request for this. However, I think users do not often have:
- a multi-repo workspace
- a PR checked out in more than one repo
so such a feature request is not very high priority.

**Comment by brianjmurrell** [comment]
Can I "un"-checkout a PR in one of the repos?

**Comment by alexr00** [comment]
@brianjmurrell to un-checkout a PR you can checkout a branch that doesn't have a PR created from it, usually your default branch is what you'd use.

If you're already on your default branch, meaning that you have a PR from your default branch to another branch, then you can add your default branch to the exclude setting. For example:
```
"githubPullRequests.ignoredPullRequestBranches": [
  "main"
]
```

Finally, if you don't want to ignore all PRs on this branch, but you do want to ignore all closed PRs, you could set:
```
	"githubPullRequests.useReviewMode": {
		"closed": false,
		"merged": false
	},
```

**Comment by Bearwolves** [comment]
> > Could the sidebar view have expanders for the checked out PRs in the same was as the source view, and many other views have?
> 
> Yes we could do something like this. I can re-open this issue and re-purpose it as a feature request for this. However, I think users do not often have:
> 
> * a multi-repo workspace
> * a PR checked out in more than one repo
> 
> so such a feature request is not very high priority.

We also have the multi-repo workspace (~80 repos) and the extension notify me due to rate limiting and then crash. It would be nice to select which repository should be currently fetched as in the `Source Control View` :)
Should I create a new feature request ticket for that or this ticket fits well?

**Comment by alexr00** [comment]
> extension notify me due to rate limiting and then crash. 

This is a separate issue. Please do open a new issue for it and add share the exact error you're getting!

**Comment by alexr00** [comment]
Closing as out of scope since we don't plan to address this.
